enos/provider/g5k.py

```diff
@@ -218,56 +269,96 @@ def _check_nodes(self,
+                break
+            
+        if len(jobs) == 0:
+            self.gridjobs = None
+        elif len(jobs) == len(queues):
+            self.gridjobs = jobs
+        else:
+            logging.error("There are some missing jobs. I found %s" % str(jobs))
+            sys.exit(21)
+        
+        return self.gridjobs
+
+    
+    def _get_jobs(self):
+        """Get the hosts from existing jobs (if any) or from new jobs.
+        This will perform reservations if necessary."""
 
         # Look if there is a running job or make a new reservation
-        self.gridjob, _ = EX5.planning.get_job_by_name(self.config['name'])
+        self._get_job_ids()
 
-        if self.gridjob is None:
+        if self.gridjobs is None:
             self._make_reservation()
         else:
-            logging.info("Using running oargrid job %s" % self.gridjob)
+            logging.info("Using running oargrid jobs %s" % self.gridjobs)
 
-        # Wait for the job to start
-        EX5.wait_oargrid_job_start(self.gridjob)
+        # Wait for the jobs to start
+        nodes = []
+        for job_id in self.gridjobs:
+            EX5.wait_oargrid_job_start(job_id)
+            nodes += EX5.get_oargrid_job_nodes(job_id)
 
-        self.nodes = sorted(EX5.get_oargrid_job_nodes(self.gridjob),
-                            key=lambda n: n.address)
+        self.nodes = sorted(nodes, key=lambda n: n.address)
+        logging.info(nodes)
 
         # XXX check already done into `_deploy`.
-        self._check_nodes(nodes=self.nodes,
+        self._check_nodes(nodes=nodes,
                           resources=self.config['resources'],
                           mode=self.config['role_distribution'])
 
-        # XXX(Ad_rien_) Start_date is never used, deadcode? - August
-        # 11th 2016
-        self.start_date = None
-        job_info = EX5.get_oargrid_job_info(self.gridjob)
-        if 'start_date' in job_info:
-            self.start_date = job_info['start_date']
-
         # filling some information about the jobs here
         self.user = None
-        job_info = EX5.get_oargrid_job_info(self.gridjob)
+        # FIXME is this correct?
+        job_info = EX5.get_oargrid_job_info(self.gridjobs[0])
         if 'user' in job_info:
             self.user = job_info['user']
 
         # vlans information
-        job_sites = EX5.get_oargrid_job_oar_jobs(self.gridjob)
+        job_sites = []
+        for job_id in self.gridjobs:
+            job_sites.append(EX5.get_oargrid_job_oar_jobs(job_id))
+            
+        job_sites = [item for sublist in job_sites for item in sublist]
+        
         self.jobs = []
         self.vlans = []
         for (job_id, site) in job_sites:
             self.jobs.append((site, job_id))
-            vlan_id = EX5.get_oar_job_kavlan(job_id, site)
-            if vlan_id is not None:
-                self.vlans.append((site,
-                                   EX5.get_oar_job_kavlan(job_id, site)))
+
```

Comment:
I don't get why you rewrite the following snippet. I see nothing specific to multiple queues. Can you enlighten me on that?